ui/components/connections/metadata.js

```diff
@@ -217,8 +176,20 @@ const KubernetesMetadataFormatter = ({ meshsyncControllerState, connection, meta
                     </ListItem>
                   </List>
                 </Grid2>
-                <Grid2 size={{ xs: 12, md: 5 }}>
+                <Grid2 size={{ xs: 12, md: 4 }}>
                   <List>
+                    {(meshSyncState || natsState) && (
+                      <ListItem>
+                        <TooltipWrappedConnectionChip
+                          tooltip={meshSyncState !== DISABLED ? `Ping MeshSync` : 'Not Available'}
+                          title={'MeshSync'}
+                          status={meshSyncState}
+                          handlePing={handleMeshSyncClick}
+                          iconSrc="/static/img/meshsync.svg"
```

Comment:
@FaheemOnHub, please open an issue to see that all connection chips use icons from their connection definitions; from their models.